pkg/rule/rule_manager.go

```diff
@@ -61,19 +70,52 @@ func (r *ManagedRules) Start(ctx context.Context) error {
 			ruleData, err := r.rulesUpdateFunc()
 			if err != nil {
 				r.logger.Errorf("error with rules update function for table/chain %v/%v: %v", r.ruleTarget.table.Name, r.ruleTarget.chain.Name, err)
+
+				err = r.metrics.Count(m.Prefix("manager_loop_update_func"), 1, []string{"manager_type:rule", "success:false"}, 1)
+				if err != nil {
+					r.logger.Warnf("error sending manager_loop_update_func metric: %v", err)
+				}
+
+				continue
+			}
+
+			err = r.metrics.Count(m.Prefix("manager_loop_update_func"), 1, []string{"manager_type:rule", "success:true"}, 1)
+			if err != nil {
+				r.logger.Warnf("error sending manager_loop_update_func metric: %v", err)
 			}
 
 			flush, err := r.ruleTarget.Update(r.conn, ruleData)
```

Comment:
We should have a gauge for the length of `ruleData` since that represents the number of rules added to the chain.

Reply:
I kind of forgot that we also have a genDelta for rules so I just updated the return values to give us added and deleted like for sets. We don't have access to the length of the set/number of rules at this level directly, we'd have to make another RPC to get it I think, or we could add it as a return value to the UpdateElements methods.

Reply:
This is better than I was thinking, thanks!